src/vcpkg/platform-expression.cpp

```diff
@@ -543,9 +545,12 @@ namespace vcpkg::PlatformExpression
                             // For backwards compatability arm is also true for arm64.
                             // This is because it previously was only checking for a substring.
                             return true_if_exists_and_equal("VCPKG_TARGET_ARCHITECTURE", "arm") ||
-                                   true_if_exists_and_equal("VCPKG_TARGET_ARCHITECTURE", "arm64");
+                                   true_if_exists_and_equal("VCPKG_TARGET_ARCHITECTURE", "arm64") ||
+                                   true_if_exists_and_equal("VCPKG_TARGET_ARCHITECTURE", "arm64ec");
```

Comment:
```suggestion
                                   true_if_exists_and_equal("VCPKG_TARGET_ARCHITECTURE", "arm64");
```

I also like @kobykahane 's suggestion because it avoids furthering the backcompat thing described in 545.